src/runtime/mod.rs

```diff
@@ -85,29 +87,44 @@ impl Runtime {
                 loop {
                     // Wait for read-readiness
                     let mut guard = driver.readable().await.unwrap();
-                    CONTEXT.with(|cx| cx.with_driver_mut(|driver| driver.tick()));
+                    CONTEXT.with(|cx| cx.with_handle_mut(|driver| driver.tick()));
                     guard.clear_ready();
                 }
             }
         };
 
         local.spawn_local(drive);
 
-        Ok(Runtime { local, rt })
+        Ok(Runtime { local, rt, driver })
     }
 
     /// Runs a future to completion on the current runtime
     pub fn block_on<F>(&self, future: F) -> F::Output
     where
         F: Future,
     {
+        struct ContextGuard;
```

Comment:
It's public because there were requests to make it public. I forget why. I actually agree that it should be public ideally, but I think that we made it public prematurely. The current API feels very rough and messy.